base/src/com/google/idea/blaze/base/sync/projectstructure/ModuleEditorImpl.java

```diff
@@ -118,8 +118,9 @@ public Module findModule(String moduleName) {
   public void commitWithGc(BlazeContext context) {
     List<Module> orphanModules = Lists.newArrayList();
     for (Module module : ModuleManager.getInstance(project).getModules()) {
-      if (!modules.containsKey(module.getName())) {
-        orphanModules.add(module);
+      if (!modules.containsKey(module.getName())
+              && (ModuleValidator.EP_NAME.getExtensionList().isEmpty() || !ModuleValidator.EP_NAME.getExtensionList().stream().anyMatch(n -> n.hasOtherContentRoot(module)))) {
```

Comment:
Do we need `isEmpty` if we already have `!...anyMatch`?

Reply:
huh..may be not. I added explicitly to handle default case, when there are no EP implementations.